src/main/kotlin/frc/robot/subsystems/vision/PhotonVisionIOReal.kt

```diff
@@ -25,28 +25,32 @@ class PhotonVisionIOReal(private val camera: PhotonCamera, private val robotToCa
     }
 
     override fun updateInputs() {
-        val latestResult = camera.latestResult
+        val unreadResults = camera.allUnreadResults
 
-        if (!latestResult.hasTargets()) {
-            return
-        }
+        lateinit var estimatedPose: Optional<EstimatedRobotPose>
 
-        val estimatedPose = estimator.update(latestResult)
+        for (result in unreadResults) {
+            if (!result.hasTargets()) {
+                return
+            }
 
-        if (estimatedPose.isEmpty) {
-            return
-        }
+            estimatedPose = estimator.update(result)
 
-        val tags = latestResult.targets
+            if (estimatedPose.isEmpty) {
+                return
+            }
 
-        inputs.distanceToTargets.clear()
-        inputs.poseFieldOriented = estimatedPose.get().estimatedPose
+            val tags = result.targets
 
-        inputs.timestamp = estimatedPose.get().timestampSeconds
-        for (tag in tags) {
-            val distanceToTarget = tag.bestCameraToTarget.translation.norm
-            inputs.distanceToTargets.add(distanceToTarget)
-        }
+            inputs.distanceToTargets.clear()
+            inputs.poseFieldOriented = estimatedPose.get().estimatedPose
 
+            inputs.timestamp = estimatedPose.get().timestampSeconds
+
+            for (tag in tags) {
+                val distanceToTarget = tag.bestCameraToTarget.translation.norm
+                inputs.distanceToTargets.add(distanceToTarget)
+            }
```

Comment:
You lose information here. What type is `tag`? Why not log at least `bestCameraToTarget`?

Reply:
We don't want to overlog, meaning slowing down the vision and causing potential loop overruns

Reply:
Logging `tag.bestCameraToTarget.translation.norm` is faster than logging ``tag.bestCameraToTarget`?

Reply:
d91a9df6e3b57c0dba37d88c2302a79fbf4507af

Reply:
It's not related

Reply:
c0d84b275b302a922dff9598e0579341f43c3f75
Forgot to push the commit